flink-runtime/src/main/java/org/apache/flink/runtime/state/StateObject.java

```diff
@@ -47,7 +49,7 @@
 	 * <p>If the the size is not known, return {@code 0}.
 	 *
 	 * @return Size of the state in bytes.
-	 * @throws Exception If the operation fails during size retrieval.
+	 * @throws IOException If the operation fails during size retrieval.
 	 */
-	long getStateSize() throws Exception;
+	long getStateSize() throws IOException;
```

Comment:
I think with the change in `StreamStateHandle`, even throwing IOException becomes obsolete now for all existing implementations. We might remove it.